src/firstboot/firstboot.c

```diff
@@ -1120,6 +1230,12 @@ static int parse_argv(int argc, char *argv[]) {
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                        "--delete-root-password cannot be combined with other root password options");
 
+        if (arg_locale_gen) {
+                r = must_be_root(); /* chroot() */
```

Comment:
Please don't. In almost all cases must_be_root() is counterproductive. See 569ad251adde02dc0915758fe027e0346e50738a for some discussion.